model/load_balancer.rb

```diff
@@ -26,6 +25,10 @@ def path
     "/location/#{private_subnet.display_location}/load-balancer/#{name}"
   end
 
+  def active_vms
+    vms_to_dns.select { |vm| vm.load_balancers_vms.node_state == "up" }
```

Comment:
Does something like this work/make sense? Rather than loading up associations only to do filtering and find one record?

```ruby
VmToDns.join(:load_balancer_vm, {vm_id: :id, node_state: 'up'}).select_all(:vm_to_dns)
```

Reply:
This is the issue we observed in prod multiple times: A health probe reports unavailable for a short duration and then available again with health probes succeeding continuously, but the load balancer doesn't route traffic. A manual `incr_update_load_balancer` on the load balancer fixes it. 

Is it possible that the logic in `Vnet::UpdateLoadBalancerNode` acts on a cached state where the value of column `state` is still `down` while in the DB it's already `up`?

Reply:
[Here](https://github.com/ubicloud/ubicloud/blob/2c0ed37580380bc12779ce0f4a09b3f5596d3993/prog/vnet/load_balancer_health_probes.rb#L28) we set `incr_update_load_balancer` when the probe succeeds again. Threshold is 1 in this case.

Which triggers: https://github.com/ubicloud/ubicloud/blob/2c0ed37580380bc12779ce0f4a09b3f5596d3993/prog/vnet/update_load_balancer_node.rb#L25
https://github.com/ubicloud/ubicloud/blob/2c0ed37580380bc12779ce0f4a09b3f5596d3993/prog/vnet/update_load_balancer_node.rb#L37

Reply:
thing is, I don't understand how this problem is different than how we smooth flaps for postgres (but maybe it is), so I wonder why we have a second approach, unless this the problem is different (possible) or the solution is superior (also possible).

It's also a little aberrant that the `monitor` daemon doesn't do the health checks, which is something we should correct. I'm not sure if we should correct it for _this_ patch, but, since Postgres has a prior art that comes from a very old and long-time adequate place (see `aggregate_readings`), I am wondering if the divergence is necessary.